### ISSUE TITLE
Memory access fault

### ISSUE THREAD

**Issue by bsobol** [AdaptiveCpp/AdaptiveCpp#332]
Hello, I have hipSYCL compiled against Clang 10, ROCm 3.8.0, and boost 1.71 on Ubuntu 20.04 with Radeon RX470 (gfx803)

I get an error:

> Memory access fault by GPU node-1 (Agent handle: 0x14d2b50) on address 0xf800000000. Reason: Page not present or supervisor privilege.

when sylccc optimisations are disabled and it turns into 

> free(): invalid pointer 

with optimisations enabled on basically everything I try to run
 
It happens on both develop and stable branches.
I've tried also ROCm 3.5.1 with no luck.

I don't know if I'm right posting issues here as there were/are a lot of issues with rocm-gfx803 combination, but maybe I'm just missing something simple.

**Comment by illuhad** [comment]
Yes, usually those kinds of issues are caused by ROCm bugs... I've heard of lots of issues with gfx8 GPUs. Maybe you could try a newer clang, or just to be sure, try our binary packages from the package repositories.

> free(): invalid pointer

This error also appears both with `stable` and `develop` branches? Are you using SYCL 2020 USM?

My current AMD test setup is clang 10 + ROCm 3.8, so that combination is fine in principle - I'm using a Radeon VII though.

**Comment by illuhad** [comment]
Not sure if this has something to do with it, but I've always wanted to test the following in case I encountered one of those issues again:

Replace `hipConfigureCall` with `__hipPushCallConfiguration` similar to CUDA a couple lines above in
https://github.com/illuhad/hipSYCL/blob/df7026b57fe9534711838e35e1c59d2815cc4666/include/hipSYCL/glue/generic/hiplike/clang.hpp#L67

This switches to the new HIP kernel configuration API - who knows, maybe some ROCm distributions don't support the older `hipConfigureCall` anymore..

**Comment by bsobol** [comment]
Yep, both branches give the same result. I'm not using USM. I'm not trying anything fancy until simplest Hello World vector addition works.

With `0.8.2` binary package I had issues even with detecting the device...
Also no luck with `__hipPushCallConfiguration`. Tomorrow I will try with nightly build and/or newer clang.

Anyway, thanks for the fast response.

**Comment by illuhad** [comment]
Okay, thanks for the feedback. Here's a very similar issue on gfx803 that apparently was solved by switching to clang 10 from apt.llvm.org:
https://github.com/illuhad/hipSYCL/issues/201#issuecomment-621480624
